blog/2019-12-18-HowToPlanAndOptimizeASecretSanta.adoc

```diff
@@ -0,0 +1,102 @@
+= How to Plan (and Optimize) a Secret Santa
+:page-interpolate: true
+:awestruct-author: cchianel
+:awestruct-layout: blogPostBase
+:awestruct-tags: [useCase]
+
+Many businesses host a "Secret Santa", where each employee get assigned
+a coworker whom they need to buy a present for. This fosters good relations
+between the employees and bring them closer together. But what about global
+companies, where coworkers may be many miles apart? Ideally, we want
+employees who are further apart to give gifts to each other, since they
+are the ones who probably know the least about each other. Let optimize
+it with OptaPlanner!
+
+== The Constraints
+
+One of the most obvious constraint is that everyone should get a gift.
+Or phrased a little differently: no one should get multiple gifts.
+
+Here is how one would create this constraint in OptaPlanner using the
+constraint streams API:
+[source,java]
+----
+private Constraint sameRecieverConflict(ConstraintFactory constraintFactory) {
+    return constraintFactory
+            .fromUniquePair(SecretSantaAssignment.class,
+                    Joiners.equal(SecretSantaAssignment::getReciever))
+            .penalize("Same Reciever", HardMediumSoftBigDecimalScore.ONE_HARD);
+}
+----
+
+Next, we need to to award assignments where the gifter and the reciever
```

Comment:
Should be `we need to award assignments where the gifter and the receiver`

Just a suggestion, but `giver and receiver` sounds more natural, but feel free to disagree.

Reply:
I feel gifter more apt, since it describes what you are giving, a gift. I don't think there is a word for "one who receives gifts" though.

Reply:
I think the better term for the person receiving a gift is actually recipient.

Reply:
recipient is used more for awards than gifts, although I think it better than receiver

Reply:
Apparently the official word for one getting a gift is.... "giftee"

Reply:
Interesting. Not sure I've ever heard the world "giftee" used. Learn something new every day.